credentials/src/test/kotlin/SSICredStatusListTest.kt

```diff
@@ -0,0 +1,277 @@
+      mapOf("firstName" to "Bobby"),
+      "123"
+    )
+
+    val testCred2 = createTestCredWithStatus(
+      did,
+      "test-subject-id-2",
+      mapOf("firstName" to "Joe"),
+      "124"
+    )
+
+    val statusListCredentialJwt =
+      VerifiableCredential.generateStatusList2021Credential(
+        signOptions,
+        revocationID,
+        testIssuer,
+        "revocation",
+        listOf(testCred1, testCred2)
+      )
+
+    val statusListCredential = VerifiableCredential.fromJwt(statusListCredentialJwt)
+
+    assertNotNull(statusListCredential)
+    assertTrue(
+      statusListCredential.contexts.containsAll(
+        listOf(
+          URI.create("https://www.w3.org/2018/credentials/v1"),
+          URI.create("https://w3id.org/vc/status-list/2021/v1")
+        )
+      )
+    )
+    assertTrue(
+      statusListCredential.types.containsAll(
+        listOf(
+          "VerifiableCredential",
+          "StatusList2021Credential"
+        )
+      )
+    )
+    assertEquals(statusListCredential.credentialSubject.id, URI.create(revocationID))
+    assertEquals(statusListCredential.credentialSubject.type, "StatusList2021")
+    assertEquals(
+      "revocation",
+      statusListCredential.credentialSubject.jsonObject["statusPurpose"] as? String?
+    )
+    assertEquals(
+      "H4sIAAAAAAAA/2NgQAESAAPT1/8QAAAA",
+      statusListCredential.credentialSubject.jsonObject["encodedList"] as? String?
+    )
+  }
+
+  @Test
+  fun `fails when duplicate statusListIndex`() {
+    val revocationID = "revocation-id"
+    val testIssuer = did
+
+    val testCred1 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("company" to "Block", "website" to "https://block.xyz"),
+      "123"
+    )
+
+    val testCred2 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("company" to "Block", "website" to "https://block.xyz"),
+      "123"
+    )
+
+    val exception = assertThrows<Exception> {
+      VerifiableCredential.generateStatusList2021Credential(
+        signOptions,
+        revocationID,
+        testIssuer,
+        "revocation",
+        listOf(testCred1, testCred2)
+      )
+    }
+
+    assertTrue(
+      exception
+        .message!!.contains("duplicate entry found")
+    )
+  }
+
+  @Test
+  fun `invalid index value`() {
+    val revocationID = "revocation-id"
+    val testIssuer = did
+
+    val testCred1 = createTestCredWithStatus(
+      did,
+      "test-subject-id-1",
+      mapOf("company" to "Block", "website" to "https://block.xyz"),
+      "-1"
+    )
+
+    val exception = assertThrows<Exception> {
+      VerifiableCredential.generateStatusList2021Credential(
+        signOptions,
+        revocationID,
+        testIssuer,
+        "revocation",
+        listOf(testCred1)
+      )
+    }
+
+    assertTrue(
+      exception
+        .message!!.contains("invalid status list index")
+    )
+  }
+
+  @Test
+  fun `validate credential exists in status cred list`() {
```

Comment:
Consider using test parameterization. See https://proandroiddev.com/kotlin-unit-tests-with-parameters-e37aab2b36f6 for an example. This should simplify the test code and make it easier to grok. It would also separate the different behaviors that you're testing, into singular tests.

Reply:
Happy to do this in a future PR, our tests definitely have a lot of reuse. 
https://github.com/TBD54566975/web5-sdk-kotlin/issues/51